### PR TITLE
add publish catalog/deploy catalog app step

### PR DESCRIPTION
**Pull request by gitlawr**
Address issue: https://github.com/rancher/rancher/issues/16026
Depending on https://github.com/rancher/pipeline-tools/pull/2

### CODE REVIEW

**Comment by orangedeng** [comment]
LGTM

File: apis/management.cattle.io/v3/k8s_defaults.go [hunk 1]
```diff
@@ -83,7 +83,8 @@ var (
 			Minio:         m("minio/minio:RELEASE.2018-05-25T19-49-13Z"),
 			Registry:      m("registry:2"),
 			RegistryProxy: m("rancher/pipeline-tools:v0.1.0"),
-			KubeApply:     m("rancher/pipeline-tools:v0.1.0"),
+			//FIXME new release of the pipeline-tools image
```
Comment (StrongMonkey) [inline apis/management.cattle.io/v3/k8s_defaults.go]: @gitlawr are you going to fix it?

Comment (StrongMonkey) [reply apis/management.cattle.io/v3/k8s_defaults.go]: nvm, i saw your pipeline pr

File: apis/project.cattle.io/v3/pipeline_types.go [hunk 2]
```diff
@@ -225,6 +227,24 @@ type ApplyYamlConfig struct {
...
+	GitURL    string `json:"gitUrl,omitempty" yaml:"gitUrl,omitempty"`
+	GitBranch string `json:"gitBranch,omitempty" yaml:"gitBranch,omitempty"`
+	GitAuthor string `json:"gitAuthor,omitempty" yaml:"gitAuthor,omitempty"`
+	GitEmail  string `json:"gitEmail,omitempty" yaml:"gitEmail,omitempty"`
+}
+
+type ApplyAppConfig struct {
+	Catalog         string `json:"catalog,omitempty" yaml:"catalog,omitempty"`
+	Version         string `json:"version,omitempty" yaml:"version,omitempty"`
+	Answers         string `json:"answers,omitempty" yaml:"answers,omitempty"`
```
Comment (StrongMonkey) [inline apis/project.cattle.io/v3/pipeline_types.go]: Should answers be map[string]string? is it expected that user has to put whole json blob into answers field?

Comment (gitlawr) [reply apis/project.cattle.io/v3/pipeline_types.go]: Yes, map makes more sense.

**Comment by StrongMonkey** [comment]
LGTM

**Pull request merged**